### PR TITLE
Update the notes about the need to restart the kueue pod

### PR DESCRIPTION
**Pull request by ChristianZaccaria**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Updating the notes from this PR: https://github.com/kubernetes-sigs/kueue/pull/2685 

Notes have been previously added until https://github.com/kubernetes-sigs/kueue/issues/2414 is fixed. 
It is now fixed by this PR: https://github.com/kubernetes-sigs/kueue/pull/2574 and is being [cherry-picked to `release-0.8`](https://github.com/kubernetes-sigs/kueue/pull/2991)

Now that v0.8.1 has been released, we've updated the notes accordingly.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: site/content/en/docs/tasks/run/rayjobs.md [hunk 1]
```diff
@@ -18,13 +18,6 @@ This guide is for [batch users](/docs/tasks#batch-user) that have a basic unders
 
 2. Check [Administer cluster quotas](/docs/tasks/manage/administer_cluster_quotas) for details on the initial Kueue setup.
 
-3. See [KubeRay Installation](https://ray-project.github.io/kuberay/deploy/installation/) for installation and configuration details of KubeRay.
```
Comment (mbobrovskyi) [inline site/content/en/docs/tasks/run/rayjobs.md:21]: Should we delete it?

Comment (ChristianZaccaria) [reply site/content/en/docs/tasks/run/rayjobs.md:21]: No, one second sorry

Comment (ChristianZaccaria) [reply site/content/en/docs/tasks/run/rayjobs.md:21]: Fixed, thanks!

**Comment by mbobrovskyi** [comment]
/release-note-edit 

```release-note 
NONE
```

**Comment by mimowo** [comment]
/hołd
Until 0.8.1 is out. 

Also, we will have users of earlier versions for a while, so I'm wondering if it is better to keep the note for a while but update that it is required prior to 0.8.1. WDYT @tenzen-y @alculquicondor ?

**Comment by mimowo** [comment]
I would suggest to retitle the PR as "Update the note about the need to restart the kueue pod". And in the note say (example): "In order to use JobSet, prior to v0.8.1, you need to restart Kueue after the installation.
You can do it by running: `kubectl delete pods -lcontrol-plane=controller-manager -nkueue-system`."

The reason being we will have users on older versions for a while. I would suggest dropping the note when we release v0.9. WDYT @tenzen-y @alculquicondor ?

**Comment by alculquicondor** [comment]
SGTM

**Comment by tenzen-y** [comment]
Or we may want to treat this automatic watching mechanism as a feature. Then we mention that "this automatic CR watcher has been available since 0.9," the same as other features.

Because there are many users who use 0.7 or older versions.

**Comment by mimowo** [comment]
> Or we may want to treat this automatic watching mechanism as a feature.

It is already categorized as a bug and cherry-picked to 0.8.1 (see [notes](https://github.com/kubernetes-sigs/kueue/releases/tag/v0.8.1)).

Also, 0.9 is not released yet (and will probably take another >2w), so this way we may encourage users to already transition to 0.8.1.

**Comment by ChristianZaccaria** [comment]
@mimowo I've updated the notes and rebased. Thank you!

**Pull request merged**